cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/DataStreamsPipelineSpec.java

```diff
@@ -34,27 +34,30 @@ public class DataStreamsPipelineSpec extends PipelineSpec {
   private final long batchIntervalMillis;
   private final String extraJavaOpts;
   private final boolean stopGracefully;
-  private final boolean checkpointsDisabled;
+  private final boolean checkpointsEnabled;
   private final boolean isUnitTest;
   private final String checkpointDirectory;
   private final String pipelineId;
+  private final boolean nativeStateTrackingEnabled;
```

Comment:
Having separate flags means it is possible for both of them to be true due to some bug, even though they are supposed to be mutually exclusive.

I think it would be a little better if all the state stuff was grouped together into some StateManagementSpec (can choose a better name). In there, we could have a set of optional properties like the checkpoint directory, along with an enum for the state management mode (checkpointing or statestore). That way we can enforce that these options are mutually exclusive.